opsdroid/connector/github/tests/test_connector_github.py

```diff
@@ -42,49 +42,40 @@ def test_missing_token(caplog):
     assert "Missing auth token!" in caplog.text
 
 
+@pytest.mark.add_response(
+    "/user", "GET", get_response_path("github_user.json"), status=200
+)
 @pytest.mark.asyncio
 async def test_connect(connector, mock_api):
-    mock_api.add_response(
-        "/user", "GET", get_response_path("github_user.json"), status=200
-    )
-
-    async def test():
-        await connector.connect()
-
-        assert mock_api.called("/user")
-        assert mock_api.call_count("/user") == 1
+    await connector.connect()
 
-        request = mock_api.get_request("/user")
-        assert "Authorization" in request.headers
-        assert "abc123" in request.headers["Authorization"]
+    assert mock_api.called("/user")
+    assert mock_api.call_count("/user") == 1
 
-        # Populated by the call to /user
-        assert connector.github_username == "opsdroid"
+    request = mock_api.get_request("/user")
+    assert "Authorization" in request.headers
+    assert "abc123" in request.headers["Authorization"]
 
-    await mock_api.run_test(test)
+    # Populated by the call to /user
+    assert connector.github_username == "opsdroid"
 
 
+@pytest.mark.add_response(
+    "/user", "GET", get_response_path("github_user_bad_credentials.json"), status=401
+)
 @pytest.mark.asyncio
 async def test_connect_failure(connector, mock_api, caplog):
-    mock_api.add_response(
-        "/user",
-        "GET",
-        get_response_path("github_user_bad_credentials.json"),
-        status=401,
-    )
-
-    async def test():
-        await connector.connect()
-        assert "Bad credentials" in caplog.text
-
-    await mock_api.run_test(test)
+    await connector.connect()
+    assert "Bad credentials" in caplog.text
 
 
+@pytest.mark.add_response("/", "GET")
```

Comment:
I decided to add a "you must register a route" check, but then these tests don't actually use a route, but they still start the mock_api server, so maybe we don't want to enforce that?

Reply:
Agreed. I see you dropped that.